### PR TITLE
flock 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by printfn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I noticed there's already an [unrelated cask named `flock`](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/flock.rb). Do I need to rename this formula to disambiguate?

### CODE REVIEW

File: Formula/flock.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  homepage "https://github.com/discoteq/flock"
+  url "https://github.com/discoteq/flock/releases/download/v0.4.0/flock-0.4.0.tar.xz"
+  sha256 "01bbd497d168e9b7306f06794c57602da0f61ebd463a3210d63c1d8a0513c5cc"
+  license "ISC"
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
```
Comment (SMillerDev) [inline Formula/flock.rb:12]:
```suggestion
    system "./configure", *std_configure_args,
                          "--disable-debug",
                          "--disable-dependency-tracking",
                          "--disable-silent-rules"
```

File: Formula/flock.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    system "make", "install"
+  end
+
+  test do
+    pid = fork do
+      exec bin/"flock", "tmpfile", "sleep", "5"
+    end
+    sleep 1
+    shell_output("#{bin}/flock --nonblock tmpfile true", 1)
```
Comment (SMillerDev) [inline Formula/flock.rb:21]:
```suggestion
    assert_match "something", shell_output("#{bin}/flock --nonblock tmpfile true", 1)
```

File: Formula/flock.rb (hunk 1 above)
Comment (printfn) [reply Formula/flock.rb]: Fixed

File: Formula/flock.rb (hunk 2 above)
Comment (printfn) [reply Formula/flock.rb]: I'm now asserting that the command output is empty

**Review by SMillerDev — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@printfn, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**